### PR TITLE
fix(metrics): there is no reactor running

### PR DESCRIPTION
**Pull request by yangby-cryptape**

### CODE REVIEW

**Review by keroro520 — APPROVED** [review]

File: sync/src/synchronizer/mod.rs [hunk 1]
```diff
@@ -1769,8 +1772,9 @@ mod tests {
             );
         }
         // There may be competition between header sync and eviction, it will case assert panic
-        let mut state = peers.state.get_mut(&sync_protected_peer).unwrap();
-        synchronizer.shared().state().tip_synced(&mut state);
+        let mut state_guard = peers.state.write();
+        let state = state_guard.get_mut(&sync_protected_peer).unwrap();
+        synchronizer.shared().state().tip_synced(state);
```
Comment (zhangsoledad) [inline sync/src/synchronizer/mod.rs:1777]: introduce by
https://github.com/nervosnetwork/ckb/commit/5a1f2b255c83fb56026493bff79b7f40adb1d861

**Review by zhangsoledad — DISMISSED** [review]
Checks passed locally.

**Comment by doitian** [comment]
Try to checkout ci.yaml from develop branch to see whether it works in rc/v0.43

**Review by doitian — CHANGES_REQUESTED** [review]
Let's make CI work for 0.43 in this PR. We'll maintain 0.43 for a long time.

**Review by doitian — APPROVED** [review]

**Review by driftluo — APPROVED** [review]

**Comment by doitian** [comment]
bors r=doitian,driftluo

**Pull request merged**